### PR TITLE
fix(runner): do not exit the runner process when a sync is cancelled

### PR DESCRIPTION
**Pull request by TBonnin**
## Describe your changes

I think that at some point we were forking/spawning a new subprocess for each script execution. When aborting we will then exit the process. However we don't do that anymore and scripts simply run in a node vm in the same runner process.
This commit modifies the sync sdk so an error is thrown instead of exiting. Jobs then discard this specific error.

## Issue ticket number and link
https://linear.app/nango/issue/NAN-2195/fix-aborting-a-script-kills-the-runner-process

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/jobs/lib/execution/operations/output.ts [hunk 1]
```diff
@@ -33,25 +33,25 @@ export async function handleSuccess({ taskId, nangoProps, output }: { taskId: st
...
-}: {
-    taskId: string;
-    nangoProps: NangoProps;
-    error: {
-        type: string;
-        payload: Record<string, unknown>;
-        status: number;
-    };
-}): Promise<void> {
+export async function handleError({ taskId, nangoProps, error }: { taskId: string; nangoProps: NangoProps; error: RunnerOutputError }): Promise<void> {
```
Comment (TBonnin) [inline packages/jobs/lib/execution/operations/output.ts:36]: no real change here. Just making the type more explicit

File: packages/runner/lib/abort.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ export const abort = (taskId: string): boolean => {
     const ac = abortControllers.get(taskId);
     if (ac) {
         ac.abort();
+        abortControllers.delete(taskId);
```
Comment (TBonnin) [inline packages/runner/lib/abort.ts:8]: unrelated to this issue but I realized we were not very good citizen, not removing the abortController from the global map once it has been aborted

File: packages/jobs/lib/execution/operations/output.ts [hunk 3]
```diff
@@ -33,25 +33,25 @@ export async function handleSuccess({ taskId, nangoProps, output }: { taskId: st
...
     const formattedError = toNangoError({
         err: error,
         defaultErrorType: `${nangoProps.scriptType}_script_failure`,
         scriptName: nangoProps.syncConfig.sync_name
     });
 
+    if (error.type === 'script_aborted') {
```
Comment (TBonnin) [inline packages/jobs/lib/execution/operations/output.ts:37]: I am not a huge fan of NangoError.type being a `string` and not a union type. I tried to refactor it but it is basically opening pandora box so I gave up (for now) :(

**Review by nalanj — APPROVED** [review]

File: packages/jobs/lib/execution/operations/output.ts [hunk 4]
```diff
@@ -33,25 +33,25 @@ export async function handleSuccess({ taskId, nangoProps, output }: { taskId: st
...
+        // do nothing, the script was aborted and its state already updated
+        logger.info(`Script was aborted. Ignoring output.`, {
+            taskId,
+            syncId: nangoProps.syncId,
+            environmentId: nangoProps.environmentId,
+            providerConfigKey: nangoProps.providerConfigKey,
+            connectionId: nangoProps.connectionId
+        });
+        return;
+    }
```
Comment (bodinsamuel) [inline packages/jobs/lib/execution/operations/output.ts:47]: Could be placed before formattedError

**Review by bodinsamuel — APPROVED** [review]

**Pull request merged**